var/spack/repos/builtin/packages/qt/qt59-qtbase-qtconfig256.patch

```diff
@@ -0,0 +1,18 @@
+diff --git a/configure.pri b/configure.pri
+index be1b166080..13fdf0b91e 100644
+--- a/configure.pri
++++ b/configure.pri
+@@ -796,10 +796,10 @@ defineTest(qtConfOutput_preparePaths) {
+         "static const char qt_configure_installation     [12+11]  = \"qt_instdate=2012-12-20\";" \
+         "" \
+         "/* Installation Info */" \
+-        "static const char qt_configure_prefix_path_str  [12+256] = \"qt_prfxpath=$$config.input.prefix\";" \
++        "static const char qt_configure_prefix_path_str  [12+1024] = \"qt_prfxpath=$$config.input.prefix\";" \
+         "$${LITERAL_HASH}ifdef QT_BUILD_QMAKE" \
+-        "static const char qt_configure_ext_prefix_path_str   [12+256] = \"qt_epfxpath=$$config.input.extprefix\";" \
+-        "static const char qt_configure_host_prefix_path_str  [12+256] = \"qt_hpfxpath=$$config.input.hostprefix\";" \
++        "static const char qt_configure_ext_prefix_path_str   [12+1024] = \"qt_epfxpath=$$config.input.extprefix\";" \
++        "static const char qt_configure_host_prefix_path_str  [12+1024] = \"qt_hpfxpath=$$config.input.hostprefix\";" \
```

Comment:
Dumb question perhaps, but why don't they (and/or this patch) just let the string be sized automatically? I.e.:
```suggestion
+        "static const char qt_configure_host_prefix_path_str  [] = \"qt_hpfxpath=$$config.input.hostprefix\";" \
```

Reply:
I thought the same thing but decided to just stick with the hard-coded length as they currently do it.  My reasoning was that if some other code that I'm missing here is relying on a fixed size 256 for this that it would be less invasive to change them to 1024 rather than `sizeof(qt_configure_host_prefix_path_str)`.  I.e. I was going to the least invasive / minimum change possible.  I absolutely agree though that for an upstream fix the hard-coded size should be removed though and not just increased.